src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden) {
+    super(name, health);
+    this.hidden = hidden || false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden) {
+    super(name, health, hidden);
```

Comment:
```suggestion
  constructor(name, health = 100) {
    super(name, health);
```
As AI mentor noted, Carnivore class doesn't  need the "hidden" parameter